flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -260,16 +262,26 @@ public void submitSQLJob(SQLJobSubmission job) throws IOException {
 	}
 
 	public void copyOptJarsToLib(String jarNamePrefix) throws FileNotFoundException, IOException {
-		final Optional<Path> reporterJarOptional;
-		try (Stream<Path> logFiles = Files.walk(opt)) {
-			reporterJarOptional = logFiles
+		copyOptJars(jarNamePrefix, lib);
+	}
+
+	public void copyOptJarsToPlugins(String jarNamePrefix) throws FileNotFoundException, IOException {
```

Comment:
But this is not really "copying jars", right? It will actually move the file from opt to lib or plugins. The problem is that one of the cases I would like to test required the jar to be actually copied to both.

Reply:
> and modify FlinkDistribution#moveJar to handle this location appropriately.
- could you clarify?

Reply:
I assume you mean to modify FlinkDistribution#mapJarLocationToPath ?

Reply:
hmmm that's a bit frustrating; now we have to open up the API to allow things that people shouldn't be doing; though admittedly we should be able to test everything a user _might_ do.

A dumb intermediate workaround would be to have 2 variants; one for copying/moving each.

You will also have to modify `#moveJar` to pass in the file(name at least) to `#mapJarLocationPath` so that you can introduce the directory for the individual plugin.